src/dec/stream_decode.cc

```diff
@@ -34,20 +35,34 @@ NAN_METHOD(StreamDecode::Transform) {
   StreamDecode* obj = ObjectWrap::Unwrap<StreamDecode>(info.Holder());
 
   Local<Object> buffer = info[0]->ToObject();
-  obj->input = BrotliInitMemInput(
-    (const uint8_t*) node::Buffer::Data(buffer),
-    node::Buffer::Length(buffer),
-    &obj->mem_input);
+  obj->next_in = (const uint8_t*) node::Buffer::Data(buffer);
+  obj->available_in = node::Buffer::Length(buffer);
 
   Nan::Callback *callback = new Nan::Callback(info[1].As<Function>());
-  Nan::AsyncQueueWorker(new StreamDecodeWorker(callback, obj, false));
+  StreamDecodeWorker *worker = new StreamDecodeWorker(callback, obj);
+  if (info[2]->BooleanValue()) {
```

Comment:
I think it might be wiser to pass the sync parameter to the `StreamDecode` and `StreamEncode` constructors instead, so we don't have to do the JS->C dance on each call.